infrastructure/cli/environment.go

```diff
@@ -0,0 +1,33 @@
+package cli
+
+import "github.com/snyk/snyk-ls/application/config"
+
+// Returns the input array with additional variables used in the CLI run in the form of "key=value".
+// Since we append, our values are overwriting existing env variables (because exec.Cmd.Env chooses the last value
+// in case of key duplications).
+func AppendCliEnvironmentVariables(currentEnv []string) (updatedEnv []string) {
```

Comment:
are/should this fn be tested for all cases?

Reply:
Fair point! I'll add tests around the func 👍

Reply:
In fact there were few tests already, I've moved them over. Also remembered that the reason why we had two different places for environment passing was because token shouldn't be passed in authentication provider. Corrected this.